routes/pasteRouter.js

```diff
@@ -15,48 +15,46 @@ routes.get(['/:documentId', '/:slug/:documentId', '/:slug/:slugTwo/:documentId',
-                        var creator;
-                        if (userId == document.creator || editorArray.indexOf(userId) != -1) creator = true;
-                        else creator = false;
-                        res.render('pasted.ejs', { documentName: documentId, imageEmbed: enableImageEmbed, code: document.code, loggedIn: true, pfp: user.icon, deleteDate: deleteDate, creator: creator, originalCreator: document.creator, incomingUser: userId })
-                    })
-                } else {
-                    res.render('pasted.ejs', { documentName: documentId, imageEmbed: enableImageEmbed, code: document.code, loggedIn: false, deleteDate: deleteDate, creator: false })
+            if(err) return res.render('error.ejs', { error: 'An error occurred whilst getting that document!' });
+            if(!document) return res.render('error.ejs', { error: 'We couldn\'t find that document!' });
+
+            if(document.instantDelete) {
+                if(!req.isCrawler()) {
+                    setTimeout(() => {
+                        db.link.remove({ URL: document.URL });
+                    }, 1 * 1000);
                 }
+                var deleteDate = "Deletes after being viewed.";    
             } else {
-                res.render('error.ejs', { error: 'We couldn\'t find that document!' });
+                const documentDate = new Date(document.deleteDate);
+                const date = {
+                    year: documentDate.getFullYear(),
+                    month: documentDate.getMonth() + 1,
+                    day: documentDate.getDate() 
+                };
+                var deleteDate = `Deletes on ${date.day}/${date.month}/${date.year}.`;
             }
-        })
-    } catch (err) {
-        res.render('error.ejs', { error: 'An error occurred whilst getting that document!' });
+            var enableImageEmbed;
+            if (document.imageEmbed && fs.existsSync(`./public/assets/img/${document.URL}.jpg`)) enableImageEmbed = true;
+            else enableImageEmbed = false;
+
+            if(req.isAuthenticated()) {
+                const userId = req.user.toString();
+                Users.findOne({ _id: userId }, (err, user) => {
+                    // If there's some sort of error just return the Guest paste.
+                    if(err) return res.render('pasted.ejs', { documentName: documentId, imageEmbed: enableImageEmbed, code: document.code, loggedIn: false, deleteDate: deleteDate, creator: false });
+                    
+                    var creator;
+                    if (userId == document.creator || editorArray.indexOf(userId) != -1) creator = true;
+                    else creator = false;
+                    res.render('pasted.ejs', { documentName: documentId, imageEmbed: enableImageEmbed, code: document.code, loggedIn: true, pfp: user.icon, deleteDate: deleteDate, creator: creator, originalCreator: document.creator, incomingUser: userId });
```

Comment:
Right here doesn't work, because the user data isnt being awaited on... (check next comment)

Reply:
Check comment below.